### PR TITLE
Support EL9

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

**Comment by ekohl** [comment]
This now fails to retrieve the `foreman-plugins` repository (but probably also `foreman` because https://github.com/theforeman/puppet-foreman/blob/557c952c1a519cb7126dfa6267eb1a5c4707880d/manifests/repos/yum.pp#L22 includes `release`, which isn't a thing on staging). Once we fix that, I think this will fail:
https://github.com/theforeman/puppet-foreman/blob/557c952c1a519cb7126dfa6267eb1a5c4707880d/manifests/repo.pp#L24-L31

We don't intend to consume modules in the initial release, so I think that should be patched:https://github.com/theforeman/puppet-foreman/pull/1147

**Comment by ekohl** [comment]
@evgeni & @ehelms: should we create a `release` -> `foreman` symlink on stagingyum for the time being? Then we can at least proceed with this, without having to run the entire pipeline every time. Or should we temporarily modify it to directly consume https://copr.fedorainfracloud.org/coprs/g/theforeman/foreman-nightly-staging directly instead?

**Comment by evgeni** [comment]
How about a foreman->releases sysmlink on yum.tfm as the first step to unifying the repo structure?

File: spec/setup_acceptance_node.pp [hunk 1]
```diff
@@ -1,5 +1,9 @@
-class { 'foreman::repo':
-  repo => 'nightly',
+yumrepo { 'foreman':
+  descr    => "Foreman ${repo}",
```
Comment (evgeni) [inline spec/setup_acceptance_node.pp:2]: Funny how this (undef variable) freaks out puppet 8 but not 7

Comment (ekohl) [reply spec/setup_acceptance_node.pp]: Puppet 8 has strict variables enabled by default.

**Comment by ekohl** [comment]
> How about a foreman->releases sysmlink on yum.tfm as the first step to unifying the repo structure?

I thought about the same, but I wondered about CDN caching efficiency. For staging that's less of a concern.

**Comment by evgeni** [comment]
> 136 examples, 71 failures

That's half of the suite passing. I like!

**Comment by ekohl** [comment]
And the failing ones are plugins, which makes sense since it doesn't set up that repository at all

File: spec/setup_acceptance_node.pp [hunk 2]
```diff
@@ -1,5 +1,9 @@
-class { 'foreman::repo':
-  repo => 'nightly',
+yumrepo { 'foreman':
+  descr    => 'Foreman nightly',
+  baseurl  => "https://download.copr.fedorainfracloud.org/results/@theforeman/foreman-nightly-staging/rhel-${facts['os']['release']['major']}-\$basearch/",
+  gpgcheck => '1',
+  gpgkey   => 'https://download.copr.fedorainfracloud.org/results/@theforeman/foreman-nightly-staging/pubkey.gpg',
+  enabled  => '1',
 }
```
Comment (evgeni) [inline spec/setup_acceptance_node.pp:7]:
```suggestion
}

yumrepo { 'foreman-plugins':
  descr    => 'Foreman Plugins nightly',
  baseurl  => "https://download.copr.fedorainfracloud.org/results/@theforeman/plugins-nightly-staging/rhel-${facts['os']['release']['major']}-\$basearch/",
  gpgcheck => '1',
  gpgkey   => 'https://download.copr.fedorainfracloud.org/results/@theforeman/plugins-nightly-staging/pubkey.gpg',
  enabled  => '1',
}
```

**Comment by evgeni** [comment]
ansible, container_gateway and pulp plugins are still missing, the rest should show up fine

**Comment by ekohl** [comment]
> mosquitto (the daemon) is missing

I submitted https://github.com/voxpupuli/puppet-mosquitto/pull/45 but realized they pull mosquitto from EPEL.

**Comment by evgeni** [comment]
136 examples, 8 failures

all ansible

TIL: we do not have smart_proxy_container_gateway tests? :D

**Comment by evgeni** [comment]
https://github.com/theforeman/puppet-foreman_proxy/pull/825

**Pull request merged**